### PR TITLE
Add processor_cpu_load_info struct and processor_cpu_load_info_t alias

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: src/unix/bsd/apple/mod.rs [hunk 1]
```diff
@@ -662,6 +664,10 @@ s_no_extra_traits! {
         __unused1: *mut ::c_void,       //actually a function pointer
         pub sigev_notify_attributes: *mut ::pthread_attr_t
     }
+
+    pub struct processor_cpu_load_info {
+        cpu_ticks: [::c_ulong; CPU_STATE_MAX],
+    }
```
Comment (JohnTitor) [inline src/unix/bsd/apple/mod.rs:670]: Failure:

```
error[E0308]: mismatched types
   --> src/unix/bsd/apple/mod.rs:669:32
    |
669 |         cpu_ticks: [::c_ulong; CPU_STATE_MAX],
    |                                ^^^^^^^^^^^^^ expected `usize`, found `i32`

error: aborting due to previous error
```

Comment (GuillaumeGomez) [reply src/unix/bsd/apple/mod.rs]: Fixed it and added a few other things. Let's see if the CI will be happier. :)

**Comment by GuillaumeGomez** [comment]
@JohnTitor CI passed. \o/

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Thanks!

r? @JohnTitor @bors r+

**Comment by stepancheg** [comment]
This PR adds definition with 32-bit fields:

```rust
    pub struct processor_cpu_load_info {
        pub cpu_ticks: [::c_uint; CPU_STATE_MAX as usize],
    }
```

However, the definition from sources uses 64-bit fields: [source](https://opensource.apple.com/source/xnu/xnu-792.12.6/osfmk/mach/processor_info.h)

```
struct processor_cpu_load_info {             /* number of ticks while running... */
        unsigned long   cpu_ticks[CPU_STATE_MAX]; /* ... in the given mode */
};
```

This is inconsistent, not sure whether the error is in Rust or in Apple source.

**Pull request merged**